### PR TITLE
cli: allow to kill a stopped container and sandbox

### PR DESCRIPTION
**Pull request by bergwolf**
cri containerd calls kill on stopped sandbox and if we
fail the call, it can cause `cri stopp` command to fail
too.

Also add a few logs to track container state transitioning.

Fixes: #1084

### CODE REVIEW

File: cli/kill.go [hunk 1]
```diff
@@ -129,13 +129,13 @@ func kill(ctx context.Context, containerID, signal string, all bool) error {
...
-		return fmt.Errorf("Container %s not ready, running or paused, cannot send a signal", containerID)
-	}
+	kataLog.WithField("signal", signal).WithField("container state", status.State.State).Info("kill")
 
-	if err := vci.KillContainer(ctx, sandboxID, containerID, signum, all); err != nil {
-		return err
+	// container MUST be created, running or paused
+	if status.State.State == vc.StateReady || status.State.State == vc.StateRunning || status.State.State == vc.StatePaused {
+		if err := vci.KillContainer(ctx, sandboxID, containerID, signum, all); err != nil {
+			return err
```
Comment (lifupan) [inline cli/kill.go]: If the sandbox/container has stopped, will the following  vci.StopSandbox/vci.StopContainer return another err?

Comment (bergwolf) [reply cli/kill.go]: No, it's ignored in sandbox.stop()

**Comment by lifupan** [comment]
It seems there's something wrong with the CI. 
LGTM!

**Comment by bergwolf** [comment]
@sboeuf Good point! After taking a closer look, it is because of a tiny difference with how we handle `kill -a` compared with runc. The `-a` option is not documented in OCI runtime spec. As a result we have to follow runc behavior and do not fail on container status when `-a` is specified, see https://github.com/opencontainers/runc/blob/master/libcontainer/container_linux.go#L378

I've updated the PR to include above change. PTAL.

**Comment by egernst** [comment]
nit: should we have split into two commits?  One commit for the change in behavior and then a second commit for adding more useful logs?

File: virtcontainers/sandbox.go [hunk 2]
```diff
@@ -1569,6 +1569,11 @@ func (s *Sandbox) Stop() error {
 	span, _ := s.trace("stop")
 	defer span.Finish()
 
+	if s.state.State == StateStopped {
+		s.Logger().Info("sandbox already stopped")
+		return nil
```
Comment (jodh-intel) [inline virtcontainers/sandbox.go]: How about adding a new unit tests for this scenario, asserting `nil` is returned when state is stopped?

File: cli/kill.go [hunk 3]
```diff
@@ -129,13 +129,15 @@ func kill(ctx context.Context, containerID, signal string, all bool) error {
...
+		}
+	} else if !all {
+		return fmt.Errorf("container not running")
```
Comment (jodh-intel) [inline cli/kill.go]: Could you add a unit test for this scenario (and the one above in the `if`)?

**Comment by sboeuf** [comment]
@jodh-intel unit test has been added, if this is okay for you, please merge the PR!

**Comment by jodh-intel** [comment]
Thanks @bergwolf!

lgtm
